src/plugins/intel_gpu/src/graph/graph_optimizer/reorder_inputs.cpp

```diff
@@ -259,20 +304,22 @@ template <direction_e dir>
 reorder_cnt count_reorders_in_dir(const std::map<program_node*, format::type>& fmt_map, layout_optimizer& lo, program_node* node) {
     size_t cnt = 0;
     size_t size = 0;
-    auto sel_fmt = fmt_map.at(node);
 
     for (auto next : travel_direction_wrapper<dir>::next_nodes(node)) {
         if (!next->is_in_data_flow())
             continue;
 
-        auto next_fmt = fmt_map.at(next);
-
-        if (next_fmt == format::any ||
-            (sel_fmt != next_fmt &&
-             !lo.can_fuse_reorder(*travel_direction_wrapper<dir>::first(node, next),
-                                  *travel_direction_wrapper<dir>::second(node, next),
-                                  travel_direction_wrapper<dir>::first(sel_fmt, next_fmt),
-                                  travel_direction_wrapper<dir>::second(sel_fmt, next_fmt)))) {
+        auto predecessor = travel_direction_wrapper<dir>::first(node, next);
+        auto successor = travel_direction_wrapper<dir>::second(node, next);
+        auto first_fmt = get_target_output_format(lo, fmt_map, predecessor, successor);
+        auto second_fmt = get_target_input_format(lo, fmt_map, successor, predecessor);
+
+        if (second_fmt == format::any ||
+            (first_fmt != second_fmt &&
+             !lo.can_fuse_reorder(*predecessor,
+                                  *successor,
+                                  travel_direction_wrapper<dir>::first(first_fmt, second_fmt),
```

Comment:
It seems wrong fmt are sent into can_fuse_reorder in backward. Could you check it? Just pass firt_fmt, second_fmt are correct NOT wrapping<dir>::first/second.

Reply:
Thanks for pointing out. Fixed.